src/OrchardCore.Modules/OrchardCore.Email.Azure/Services/AzureEmailProviderBase.cs

```diff
@@ -224,4 +226,33 @@ private EmailMessage FromMailMessage(MailMessage message, Dictionary<string, ILi
 
         return emailMessage;
     }
+
+    private static EmailAddress ToAzureEmailAddress(string emailWithDisplayName)
```

Comment:
The method name should be a verb, why didn't you add this code to `ParseEmailAddressWithDisplayName`?

Reply:
> The method name should be a verb

Yeah, I initially intended it to be an extension method, I'll fix that.

> why didn't you add this code to `ParseEmailAddressWithDisplayName`?

`ParseEmailAddressWithDisplayName` is used separately too.

Reply:
One more thing can we create a virtual method so subclasses can implement how to get the display name

Reply:
The logic here is laser-focused on this scenario, so I don't see any reason to override it without overriding the whole email sending logic implementation. But then you can handle email addresses however you want.